### 抽样分布简介

为了更好地理解抽样分布的概念，让我们先用 python 语言做一些练习。

下面是一个数组，代表我们在之前的视频中看到的学员。其中，1代表喝咖啡的学生，0代表不喝咖啡的学生。

In [2]:
import numpy as np
np.random.seed(42)

students = np.array([1,0,1,1,1,1,0,0,0,0,1,1,1,1,1,1,1,1,1,1,0])

`1.` 计算出上述数组中喝咖啡的学生的比例，并将此值存储在变量 **p** 中。

`2.` 使用 numpy 的 **random.choice** 模拟从 `students` 数组中进行5次取样。喝咖啡学员样本的比例是多少？ 

`3.` 重复上述步骤，再另外获取10,000个比例值，其中每个样本的大小为5。把这些数值存储在 `sample_props` 变量中。

`4.` 这10，000个比例值的平均比例值是多少？该值通常被称为 **抽样分布均值** 。

`5.` 21个原始数据值的方差和标准差分别是多少？

`6.` 你获取的10000个比例值的方差和标准差分别是多少？

`7.` 计算 p(1-p)，你的哪个答案与它最接近？ 

`8.` 计算 p(1-p)/n，你的哪个答案与它最接近？ 

`9.` 需要注意的是， `8.` 中的答案通常被称为 **抽样分布方差** 。如果将第一个样本大小更改为20，最终的抽样分布方差会发生什么变化？ 模拟并重新计算 `6.` 与 `8.` 的答案，检验一下你之前发现的一致性是否仍然成立。

### Simulate your 5 draws

In [10]:
p = students.mean()
p

0.7142857142857143

In [45]:
sample_five = np.random.choice(students,5)
sample_five.mean()

0.6

In [42]:
sample_props = np.random.choice(students,size = (int(1e5),5))
sample_props.mean()

0.713548

In [43]:
variance_original = np.var(students)
variance_original

0.20408163265306126

In [14]:
std_original = np.std(students)
std_original

0.45175395145262565

In [16]:
var_1e5 = np.var(sample_props)
var_1e5

0.20384463375599998

In [18]:
std_1e5 = np.std(sample_props)
std_1e5

0.45149156554248054

In [22]:
p*(1-p)

0.20408163265306123

+ p*(1-p) 与 21个原始数据的方差,以及10000个比例值得方差相似

In [39]:
p*(1-p)/5

0.04081632653061225

### Simulate your 20 draws

In [19]:
p = students.mean()

0.7142857142857143

In [26]:
sample_twenty = np.random.choice(students,20)
sample_twenty.mean()

0.65

In [29]:
sample_props_twenty = np.random.choice(students,size =(int(1e5),20))
sample_props_twenty.mean()

0.71424

In [30]:
var_original_twenty = np.var(students)
var_original_twenty

0.20408163265306126

In [31]:
std_original_twenty = np.std(students)
std_original_twenty

0.45175395145262565

In [32]:
var_1e5_20 = np.var(sample_proprs_twenty)
var_1e5_20

0.20410550709999997

In [34]:
std_1e5_20 = np.std(sample_proprs_twenty)
std_1e5_20

0.45178037485043543

In [35]:
p*(1-p)

0.20408163265306123

In [38]:
p*(1-p)/20

0.010204081632653062

In [ ]:
##Compare your variance values as computed in 6 and 8, 
##but with your sample of 20 values

`10.` 最后，从样本数量为5的样本和样本数量为20的样本中进行10000次取样，然后绘制一个直方图。这些分布中的每一个都是一个样本分布。 一个是样本量5的比例，另一个是样本量20的样本分布。

> 1. 每个抽样分布的中心应该与所有喝咖啡的 21 个学生比例相同。
    + 0.71428
> 2. 第一个样本中喝咖啡的 5 个学生比例。
    + 0.6
> 3. 具有喝咖啡习惯的所有 21 个学生原始比例的相关方差。
    + 0.20408
> 4. 样本大小为 5 的总体中抽样分布方差。
    + 0.0408
> 5. 样本大小为 20 的总体中抽样分布方差。
    + 0.0102 
> 6. 样本分布的平均数
    + p
> 7. 原值 1,0 的方差
    + p(1-p)
> 8. 20 个随机选择的数值计算出来的比例方差重复 10,000 次。
    + p(1-p)/20